### PR TITLE
Enable systemd allow nonroot

### PR DESCRIPTION
**Pull request by sallyom**
This PR 
* adds a simple image, to contain a script that calls systemctl commands on the host
* adds the function to systemd method to enable services, root user
* adds message for non-root users to run `systemctl --user enable service --now` on host to enable service as non-root
        * the non-root systemd startup is blocked at this time, need more work with the permissions/mounts/etc
TODO on follow-up:
* add systemd flow to stop & restart (update) services
* add logic for non-root enable

### CODE REVIEW

File: images/systemd/Dockerfile-systemd-enable [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM registry.access.redhat.com/ubi8/ubi:latest
```
Comment (cooktheryan) [inline images/systemd/Dockerfile-systemd-enable:1]: @sallyom we can probably drop these in `method_containers`

File: pkg/engine/api/types.go [hunk 2]
```diff
@@ -29,10 +29,15 @@ type Raw struct {
 
 type Systemd struct {
 	TargetPath string `mapstructure:"targetPath"`
-	User       string `mapstructure:"user"`
-	Schedule   string `mapstructure:"schedule"`
-	InitialRun bool
-	LastCommit *object.Commit
+	Root       bool   `mapstructure:"root"`
+	// If non-root, home directory must be supplied to place unit file at $HOME/.config/systemd/user
```
Comment (cooktheryan) [inline pkg/engine/api/types.go:33]: @sallyom should we do the equivalent of a mkdir -p just to ensure the path exists? I know its extra work on us but I dont think the subdirectory exists by default

Comment (cooktheryan) [reply pkg/engine/api/types.go:33]: we may also have to do a chown as well or set permissions to be 777

Comment (cooktheryan) [reply pkg/engine/api/types.go:33]: that would be just for the non root users

File: examples/systemd-non-root-enable.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+volume: harpoon-volume # this needs to match the volume name passed to the podman run command
+targets:
+- name: harpoon
+  url: http://github.com/sallyom/harpoon
+  systemd:
+    targetPath: examples/systemd
+    root: false
+    nonRootHomeDir: "/home/somalley"
```
Comment (sallyom) [inline examples/systemd-non-root-enable.yaml:8]: * will update this with the nonroot-homedir of the ubuntu instance

File: pkg/engine/api/types.go (hunk 2 above)
Comment (sallyom) [reply pkg/engine/api/types.go:33]: I _think_ in order for homeDir to be non-root, host podman would be non-root socket, which means that directory will exist? I see `~/.config/systemd/user/sockets.target.wants/podman.socket` on my system and am assuming the act of `systemctl --user enable podman.socket` creates that if it doesn't exist. wdyt?

**Comment by cooktheryan** [comment]
@sallyom do you want to create the GH actions for building the image, archive, then load into the systemd test?

File: pkg/engine/api/types.go (hunk 2 above)
Comment (cooktheryan) [reply pkg/engine/api/types.go:33]: Ah yes good call.

File: examples/systemd-non-root-enable.yaml (hunk 3 above)
Comment (sallyom) [reply examples/systemd-non-root-enable.yaml:8]: @cooktheryan what's the non-root homedir of the ubuntu instance in gh actions? :)

**Comment by sallyom** [comment]
> @sallyom do you want to create the GH actions for building the image, archive, then load into the systemd test?

I added the systemd-enable test, is that what you mean here?

**Comment by sallyom** [comment]
> > @sallyom do you want to create the GH actions for building the image, archive, then load into the systemd test?
> 
> I added the systemd-enable test, is that what you mean here? edit: you mean the new systemd-script! got it, will do that - we should wait to merge this til that is in, probably

Added the cross-build targets for `quay.io/harpoon/harpoon-systemd:latest` 
(`quay.io/harpoon/harpoon-systemd-amd:latest`, `quay.io/harpoon/harpoon-systemd-arm:latest`)

**Comment by cooktheryan** [comment]
perfect that was exactly what i meant :D

File: .github/workflows/docker-image.yml [hunk 4]
```diff
@@ -265,7 +312,46 @@ jobs:
...
+        with:
+          name: harpoon-image
+          path: /tmp
+
+      - name: Load the image
+        run: sudo podman load -i /tmp/harpoon.tar
+
+      - name: tag the image
+        run: sudo podman tag quay.io/harpoon/harpoon-amd:latest quay.io/harpoon/harpoon:latest
+
```
Comment (cooktheryan) [inline .github/workflows/docker-image.yml:336]: we will need to do the same steps of loading the systemd image and tagging it as harpoon-systemd:latest here that way we test against the newly created image

Comment (cooktheryan) [reply .github/workflows/docker-image.yml:336]: that should allow for us to pass the failing test

File: pkg/engine/systemd.go [hunk 5]
```diff
@@ -12,7 +17,57 @@ func systemdPodman(ctx context.Context, mo *FileMountOptions) error {
...
+		klog.Info("At this time, harpoon non-root user cannot enable systemd service on the host")
+		klog.Infof("To enable this non-root service, run 'systemctl --user enable %s' on host machine", service)
+		klog.Info("To enable service as root, run with Systemd.Root = true")
+		return nil
+	}
+
+	os.Setenv("SERVICE", service)
+	s := specgen.NewSpecGenerator(systemdImage, false)
+	runMount := "/run/systemd"
+	if !root {
```
Comment (cooktheryan) [inline pkg/engine/systemd.go:45]: We may need an enhancement request for figuring out the uid dynamically for the user but it should not stop this PR from merging

**Pull request merged**